### PR TITLE
201 private endpoint sql from webapp

### PR DESCRIPTION
**Pull request by mblanco77**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    "subnetAppServicePrefix": "10.0.1.0/24",
+    "subnetAppServiceName": "subnetAppService",
+    "sqlServerName": "[concat('sqlserver', uniqueString(resourceGroup().id))]",
+    "databaseName": "[concat(variables('sqlServerName'),'/sample-db')]",
+    "privateEndpointName": "myPrivateEndpoint",
+    "privateDnsZoneName": "privatelink.database.windows.net",
+    "pvtendpointdnsgroupname": "[concat(variables('privateEndpointName'),'/mydnsgroupname')]",
+    "websitename": "[take(concat('webapppvl', uniqueString(resourceGroup().id)),15)]",
+    "appServicePlanName": "[take(concat('appSrvPln', uniqueString(resourceGroup().id)),15)]",
+    "sqlConnString": "[concat('Server=tcp:',variables('sqlServerName'), '.database.windows.net,1433;Initial Catalog=sample-db;Persist Security Info=False;User ID=', parameters('sqlAdministratorLogin'), ';Password=', parameters('sqlAdministratorLoginPassword'), ';MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;')]"
```
Comment (bmoore-msft) [inline 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json:54]: use referencec() to get the fqdn

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+    "vnetName": "myVirtualNetwork",
+    "vnetAddressPrefix": "10.0.0.0/16",
+    "subnet1Prefix": "10.0.0.0/24",
+    "subnet1Name": "mySubnet",
+    "subnetAppServicePrefix": "10.0.1.0/24",
+    "subnetAppServiceName": "subnetAppService",
+    "sqlServerName": "[concat('sqlserver', uniqueString(resourceGroup().id))]",
+    "databaseName": "[concat(variables('sqlServerName'),'/sample-db')]",
+    "privateEndpointName": "myPrivateEndpoint",
+    "privateDnsZoneName": "privatelink.database.windows.net",
```
Comment (bmoore-msft) [inline 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json:50]: use referencec() to get the fqdn

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+        "displayName": "[variables('sqlServerName')]"
+      },
+      "kind": "v12.0",
+      "properties": {
+        "administratorLogin": "[parameters('sqlAdministratorLogin')]",
+        "administratorLoginPassword": "[parameters('sqlAdministratorLoginPassword')]",
+        "version": "12.0",
+        "publicNetworkAccess": "Disabled"
+      },
+      "resources": [
```
Comment (bmoore-msft) [inline 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json:72]: remove this property

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+        "name": "[variables('appServicePlanName')]"
+      }
+    },
+    {
+      "name": "[variables('websitename')]",
+      "type": "Microsoft.Web/sites",
+      "apiVersion": "2019-08-01",
+      "location": "[parameters('location')]",
+      "tags": {
+        "[concat('hidden-related:', resourceGroup().id, '/providers/Microsoft.Web/serverfarms/appServicePlanpvtlink')]": "Resource",
```
Comment (bmoore-msft) [inline 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json:228]: use resourceId, e.g.

[concat('hidden-related:', resourceId('Microsoft.Web/serverfarms', 'appServicePlanpvtlink')]": "Resource"

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+          "apiVersion": "2019-08-01",
+          "location": "[parameters('location')]",
+          "dependsOn": [
+            "[resourceId('Microsoft.Web/sites', variables('websitename'))]"
+          ],
+          "tags": {
+            "displayName": "Web Deploy for webapppvtlink"
+          },
+          "properties": {
+            "packageUri": "[concat(parameters('_artifactsLocation'), 'artifacts/AdventureWorks.Web.zip', parameters('_artifactsLocationSasToken'))]",
```
Comment (bmoore-msft) [inline 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json:276]: remove concat and use uri()

[uri(parameters('_artifactsLocation'), concat('artifacts/AdventureWorks.Web.zip', parameters('_artifactsLocationSasToken')))]

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json (hunk 1 above)
Comment (mblanco77) [reply 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json]: done in ab7cfb8

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json (hunk 2 above)
Comment (mblanco77) [reply 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json]: this has to be done this way

https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns#azure-services-dns-zone-configuration

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json (hunk 3 above)
Comment (mblanco77) [reply 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json]: done in ab7cfb8

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json (hunk 4 above)
Comment (mblanco77) [reply 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json]: done in ab7cfb8

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json (hunk 5 above)
Comment (mblanco77) [reply 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json]: done in ab7cfb8

File: 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json (hunk 2 above)
Comment (bmoore-msft) [reply 201-privatelink-endpoint-sql-from-appservice/azuredeploy.json]: lovely

**Pull request merged**